lib/taskana-core/src/main/java/pro/taskana/TaskanaConfiguration.java

```diff
@@ -0,0 +1,696 @@
+
+    this.roleMap =
+        builder.roleMap.entrySet().stream()
+            .collect(
+                Collectors.toUnmodifiableMap(
+                    Entry::getKey, e -> Collections.unmodifiableSet(e.getValue())));
+
+    this.securityEnabled = builder.securityEnabled;
+    this.useManagedTransactions = builder.useManagedTransactions;
+    this.domains = Collections.unmodifiableList(builder.domains);
+    this.classificationTypes = Collections.unmodifiableList(builder.classificationTypes);
+
+    this.classificationCategoriesByTypeMap =
+        builder.classificationCategoriesByTypeMap.entrySet().stream()
+            .collect(
+                Collectors.toUnmodifiableMap(
+                    Entry::getKey, e -> Collections.unmodifiableList(e.getValue())));
+
+    this.customHolidays = Collections.unmodifiableList(builder.customHolidays);
+    this.deleteHistoryOnTaskDeletionEnabled = builder.deleteHistoryOnTaskDeletionEnabled;
+    this.germanPublicHolidaysEnabled = builder.germanPublicHolidaysEnabled;
+    this.corpusChristiEnabled = builder.corpusChristiEnabled;
+    this.jobBatchSize = builder.jobBatchSize;
+    this.maxNumberOfJobRetries = builder.maxNumberOfJobRetries;
+    this.cleanupJobFirstRun = builder.cleanupJobFirstRun;
+    this.cleanupJobRunEvery = builder.cleanupJobRunEvery;
+    this.cleanupJobMinimumAge = builder.cleanupJobMinimumAge;
+    this.taskCleanupJobAllCompletedSameParentBusiness =
+        builder.taskCleanupJobAllCompletedSameParentBusiness;
+    this.validationAllowTimestampServiceLevelMismatch =
+        builder.validationAllowTimestampServiceLevelMismatch;
+    this.addAdditionalUserInfo = builder.addAdditionalUserInfo;
+    this.priorityJobBatchSize = builder.priorityJobBatchSize;
+    this.priorityJobFirstRun = builder.priorityJobFirstRun;
+    this.priorityJobRunEvery = builder.priorityJobRunEvery;
+    this.priorityJobActive = builder.priorityJobActive;
+    this.userRefreshJobRunEvery = builder.userRefreshJobRunEvery;
+    this.userRefreshJobFirstRun = builder.userRefreshJobFirstRun;
+    this.minimalPermissionsToAssignDomains =
+        Collections.unmodifiableList(builder.minimalPermissionsToAssignDomains);
+
+    if (LOGGER.isDebugEnabled()) {
+
+      StringBuilder result = new StringBuilder();
+      String newLine = System.getProperty("line.separator");
+      Field[] fields = this.getClass().getDeclaredFields();
+      result.append("TaskanaConfiguration:").append(newLine);
+      // print field names paired with their values
+      for (Field field : fields) {
+        try {
+          result.append(field.getName());
+          result.append(": ");
+          // requires access to private field:
+          result.append(field.get(this));
+        } catch (IllegalAccessException ex) {
+          System.out.println(ex);
+        }
+        result.append(newLine);
+      }
+      LOGGER.debug(result.toString());
+    }
+  }
```

Comment:
let's talk about this :) I don't like the reflection magic and would prefer the usage of a toString method. Especially when we've introduced Lombok this should be very neat.